harmonic_demo/CartPole/README.md

```diff
@@ -0,0 +1,27 @@
+# Cart-Pole Demo
+
+The Cart-Pole demo showcases how Gazebo systems can be written in Python. It implements an LQR controller, to balance 
+the Cart-Pole such that the pole is always upright.
+
+The plugin subscribes to the `cart_pole/reset_angle` topic which can be used to repeatedly reset the pole angle and let
+the controller balance the cart-pole.
+
+```
+gz topic -t "cart_pole/reset_angle" -m gz.msgs.Double -p "data: 0.3"
+```
+
+*Note*: Large angles will cause the controller to fail, in which case, the simulation has to be reset.
+
+This demo also shows that by putting the LQR controller implementation in a separate module, the plugin can provide a
+mechanism for hot reloading the controller while the simulator is running. This can be used, for example, to tune
+parameters, or quickly iterate on controller designs. To try this, modify the controller gains `R` and `Q` in `lqr_controller.py`
+and reload the controller by publishing to the `cart_pole/reload_controller` topic
+
+```
+gz topic -t "cart_pole/reload_controller" -m gz.msgs.Empty -p ""
```

Comment:
when I do this I get:

```sh
Reloading controller
terminate called after throwing an instance of 'pybind11::error_already_set'
  what():  TypeError: CartPoleSystem.init_controller() missing 2 required positional arguments: 'sdf' and 'ecm'

At:
  /home/ian/code/harmonic_demo/harmonic_demo/CartPole/plugins/cart_pole_controller/cart_pole_system.py(114): reload_controller_cb
  /home/ian/code/gz_h_ws/install/lib/python/gz/transport13/__init__.py(108): cb_deserialize
```

Any ideas?

Reply:
Fixed in https://github.com/gazebosim/harmonic_demo/pull/14/commits/56b6d571039be24995a181e9c61c2485a8f8c90b

Reply:
No longer crashes for me. I see the `Reloading controller` line printed.

At first I wasn't sure if my changes in `lqr_controller.py` did anything because it's already in a stable state. I lowered the values in these 2 lines below, reset the angle, and confirmed that it's not longer able to balance the pole.
```
        Q[0, 0] = 10
        Q[1, 1] = 10
```